### ISSUE TITLE
Disable resampling option

### ISSUE THREAD

**Issue by k9bbzJavftZaqAqSW3dY** [Taiko2k/TauonMusicBox#809]
Wonderful music player! I have no idea how you do it but this looks and feels better than what some trillion dollar companies are trying to peddle.

I remember there was an option to use phazor or gstreamer, and this option seems to have vanished recently. I actually made use of it, and changed it from phazor to gstreamer for the following reason:
- I use pipewire, and have it configured to dynamically change the hardware sound device bitrate to match the source material
- gstreamer happened to not deal with the bitrate, so that pipewire played at the accurate, non-resampled bitrate
- phazor on the other hand always made it 48kHz, and pipewire thus was forced to played the resampled sound.

Since the option is gone, I noticed pipe reporting 48kHz, i.e. I'm assuming that phazor or something similar is the default now. Is there any chance to bring back gstreamer or somehow tell phazor not to resample?

Cheers!

**Comment by k9bbzJavftZaqAqSW3dY** [comment]
From a vanilla arch pipewire setup, this is how I set up the dynamic sampling rate:
```#!/bin/bash
sudo mkdir /etc/pipewire
sudo cp /usr/share/pipewire/pipewire.conf /etc/pipewire/
sudo sed -i 's/#default.clock.allowed-rates = \[ 48000 \]/default.clock.allowed-rates = \[ 44100 48000 96000 192000 \]/g' /etc/pipewire/pipewire.conf
```

To verify what the current sampling rate is, you need pipewire-alsa, and run this:
`cat /proc/asound/card*/*/sub*/hw_params`

**Comment by k9bbzJavftZaqAqSW3dY** [comment]
Alright, I guess sometimes it is what it is. Still, nice work altogether.

**Comment by Koneko349** [comment]
This! I was wondering why my audio was begin resampled. I use an external DAC which is very very good at resampling and playing audio at their native resolutions. Please offer a way to disable the built in resampling and play audio files at the native resolution & bitrate

**Comment by Taiko2k** [comment]
OK, Ive added a setting for this now under MENU > Settings > Audio > Avoid resampling

Limitation: It only works properly for FLAC, MP3, OGG and OPUS. 

Remember, in any case you need special Pipewire config so that Pipewire wont resample either.

Flatpak [beta](https://github.com/Taiko2k/TauonMusicBox/wiki/Beta-Flatpak) should now be updated with this. Let me know if it appears to be working.

____

##### Footnotes

Resampling and downscaling will still occur for other codecs (WAV, AAC etc). Note that in Phazor, downscaling is never applied for those former codecs, regardless of this new setting, only resampling, but the others will currently always be downscaled to 44.1khz.

So basically:

 - Optimal quality: FLAC, MP3, OGG and OPUS + Avoid resampling
 - Fine quality: FLAC, MP3, OGG and OPUS
 - Maybe not best quality: WAV, M4A, APE, WMA, MOD, TTA

I need to make a table to convey this more clearly maybe.

**Comment by Koneko349** [comment]
Sorry, I am unable to test the functionality, all my audio is ALAC as M4A file.

**Comment by Taiko2k** [comment]
Ah, unfortunately this feature won't work for ALAC. I currently don't plan to move ALAC out from *category 3* as per above table.

**Comment by Taiko2k** [comment]
@yuraih This feature made to release a while ago now, forgot why I had this issue open
